spec/abstractops.html

```diff
@@ -45,59 +45,35 @@ <h1>GetOptionsObject ( _options_ )</h1>
     </p>
 
     <emu-alg>
       1. Assert: Type(_options_) is Object.
       1. Let _value_ be ? Get(_options_, _property_).
       1. If _value_ is *undefined*, return _fallback_.
-      1. Assert: _type_ is *"boolean"* or *"string"*.
-      1. If _type_ is *"boolean"*, then
-        1. Let _value_ be ! ToBoolean(_value_).
-      1. If _type_ is *"string"*, then
-        1. Let _value_ be ? ToString(_value_).
+      1. If _types_ contains Type(_value_), then
+        1. Let _type_ be Type(_value_).
+      1. Else,
+        1. Let _type_ be the last element of _types_.
+      1. Assert: _type_ is Boolean, String, or Number.
+      1. If _type_ is Boolean, then
+        1. Set _value_ to ! ToBoolean(_value_).
+      1. Else if _type_ is Number, then
+        1. Set _value_ to ? ToNumber(_value_).
+        1. If _value_ is *NaN*, throw a *RangeError* exception.
+      1. Else,
+        1. Set _value_ to ? ToString(_value_).
       1. If _values_ is not *undefined* and _values_ does not contain an element equal to _value_, throw a *RangeError* exception.
       1. Return _value_.
     </emu-alg>
   </emu-clause>
 
-  <!-- Copied from ECMA-402 DefaultNumberOption -->
-  <emu-clause id="sec-defaultnumberoption" aoid="DefaultNumberOption">
-    <h1>DefaultNumberOption ( _value_, _minimum_, _maximum_, _fallback_ )</h1>
-
-    <p>
-      The abstract operation DefaultNumberOption converts _value_ to a Number value, checks whether it is in the allowed range, and fills in a _fallback_ value if necessary.
-    </p>
-
-    <emu-alg>
-      1. If _value_ is *undefined*, return _fallback_.
-      1. Let _value_ be ? ToNumber(_value_).
-      1. If _value_ is *NaN* or less than _minimum_ or greater than _maximum_, throw a *RangeError* exception.
-      1. Return floor(_value_).
-    </emu-alg>
-  </emu-clause>
-
-  <!-- Copied from ECMA-402 GetNumberOption -->
-  <emu-clause id="sec-getnumberoption" aoid="GetNumberOption">
-    <h1>GetNumberOption ( _options_, _property_, _minimum_, _maximum_, _fallback_ )</h1>
-
-    <p>
-      The abstract operation GetNumberOption extracts the value of the property named _property_ from the provided _options_ object, converts it to a Number value, checks whether it is in the allowed range, and fills in a _fallback_ value if necessary.
-    </p>
-
-    <emu-alg>
-      1. Assert: Type(_options_) is Object.
-      1. Let _value_ be ? Get(_options_, _property_).
-      1. Return ? DefaultNumberOption(_value_, _minimum_, _maximum_, _fallback_).
-    </emu-alg>
-  </emu-clause>
-
   <emu-clause id="sec-getstringornumberoption" aoid="GetStringOrNumberOption">
     <h1>GetStringOrNumberOption ( _options_, _property_, _stringValues_, _minimum_, _maximum_, _fallback_ )</h1>
```

Comment:
there doesn't appear to be an assertion (in prose, or in the steps) that `minimum` and `maximum` is a Number, but without that assertion, the comparison to `value` would be incoherent.

Reply:
In #1413 it seems to be suggested that comparisons should be done on mathematical values, so I'll sort this out there.